spynnaker/pyNN/models/neuron/abstract_population_vertex.py

```diff
@@ -270,16 +281,18 @@ def create_machine_vertex(
             self, vertex_slice, resources_required, n_machine_time_steps,
             label=None, constraints=None):
 
-        is_recording = len(self._neuron_recorder.recording_variables) > 0 or \
-                       self._spike_recorder.record
+        is_recording = len(self._neuron_recorder.recording_variables) > 0
         buffered_sdram_per_timestep = self._get_buffered_sdram_per_timestep(
             vertex_slice)
+        buffered_sdram = self._get_buffered_sdram(
+            vertex_slice, n_machine_time_steps)
         minimum_buffer_sdram = recording_utilities.get_minimum_buffer_sdram(
-            buffered_sdram_per_timestep, n_machine_time_steps,
-            self._minimum_buffer_sdram)
+            buffered_sdram, self._minimum_buffer_sdram)
+        overflow_sdram = self._neuron_recorder.get_sampling_overflow_sdram(
+            vertex_slice)
         vertex = PopulationMachineVertex(
             resources_required, is_recording, minimum_buffer_sdram,
-            buffered_sdram_per_timestep, label, constraints)
+            buffered_sdram_per_timestep, label, constraints, overflow_sdram)
```

Comment:
If get_buffered_sdram_per_timestep is changed to get_n_timesteps_in_buffer_space, this should pass in the neuron_recorder instead of buffered_sdram_per_timestep and overflow_sdram.

Reply:
See above on why get_n_timesteps_in_buffer_space is bad!